app/models/concerns/test_track/identity.rb

```diff
@@ -46,6 +46,17 @@ def test_track_identifier(identifier_type, identifier_value_method)
             end
           end
         end
+
+        define_method :test_track_sign_up! do
+          discriminator = TestTrack::IdentitySessionDiscriminator.new(self)
+          identifier_value = send(identifier_value_method)
+
+          if discriminator.participate_in_unauthenticated_session?
+            discriminator.controller.send(:test_track_session).sign_up! identifier_type, identifier_value
+          else
+            TestTrack::OfflineSession.create_visitor_for(identifier_type, identifier_value)
```

Comment:
And now to flip flop again, I'm thinking that if you call `test_track_sign_up!` you probably expect to be creating some kind of continuity with what happened before, which is impossible given that this can be credibly replaced with a noop if not participating in an unauthenticated session.  I think I'm back in the raise camp.